update security consideration (#170)
oauth-wg/draft-ietf-oauth-status-list: draft-ietf-oauth-status-list.md

# Security Considerations {#Security}

## Correct decoding and parsing of the encoded status list

> Implementers should be particularly careful for the correct parsing and decoding of the status list. Incorrect implementations may check the index on the wrong data or misscalculate the bit and byte index leading to an erroneous status of the Referenced Token. Beware, that bits are indexed (bit order) from least signicant bit to most significant bit (also called "right to left") while bytes are indexed (byte order) in their natural incrementing byte order (usually written for display purpose from left to write). Endianess does not apply here because each status value fits within a single byte.

Reviewer: ```suggestion
Implementers should be particularly careful for the correct parsing and decoding of the status list. Incorrect implementations might check the index at the wrong position or miscalculate the bit and byte index leading to an erroneous status of the Referenced Token. Beware, that bits are indexed (bit order) from least significant bit to most significant bit (also called "right to left") while bytes are indexed (byte order) in their natural incrementing byte order (usually written for display purpose from left to write). Endianness does not apply here because each status value fits within a single byte.
```

Author: with the first phrase "on the wrong data" I meant for example calculating index on the still compressed data

Reviewer: ```suggestion
Implementers should be particularly careful for the correct parsing and decoding of the status list. Incorrect implementations might check the index at the wrong data or miscalculate the bit and byte index leading to an erroneous status of the Referenced Token. Beware, that bits are indexed (bit order) from least significant bit to most significant bit (also called "right to left") while bytes are indexed (byte order) in their natural incrementing byte order (usually written for display purpose from left to write). Endianness does not apply here because each status value fits within a single byte.
```

# Security Considerations {#Security}

## Correct decoding and parsing of the encoded status list

Implementers should be particularly careful for the correct parsing and decoding of the status list. Incorrect implementations might check the index on the wrong data or miscalculate the bit and byte index leading to an erroneous status of the Referenced Token. Beware, that bits are indexed (bit order) from least significant bit to most significant bit (also called "right to left") while bytes are indexed (byte order) in their natural incrementing byte order (usually written for display purpose from left to write). Endianness does not apply here because each status value fits within a single byte.

> Implementations shall always verify correctness using the test vectors given by this specification.

Reviewer: Should we use stronger language here? If yes, we have probably more parts in this section, where it would make sense.
```suggestion
Implementations are RECOMMENDED to verify correctness using the test vectors given by this specification.
```

## Observability of Relying Parties {#privacy-relying-party}

Once the Relying Party receives the Referenced Token, this enables him to request the Status List to validate its status through the provided `uri` parameter and look up the corresponding `index`. However, the Relying Party may persistently store the `uri` and `index` of the Referenced Token to request the Status List again at a later time. By doing so regularly, the Relying Party may create a profile of the Referenced Token's validity status. This behaviour may be intended as a feature, e.g. for a KYC process that requires regular validity checks, but might also be abused in cases where this is not intended and unknown to the Holder, e.g. profiling the suspension of a driving license or checking the employment status of an employee credential.

> TODO elaborate on status list only providing the up-to date/latest status, no historical data, may be provided by the underlying hosting architecture

Reviewer 2: Are we leaving this in or shall we move it to an issue?

Author: we had in before, so I would leave it in here

# Security Considerations {#Security}

## Correct decoding and parsing of the encoded status list

Implementers should be particularly careful for the correct parsing and decoding of the status list. Incorrect implementations might check the index on the wrong data or miscalculate the bit and byte index leading to an erroneous status of the Referenced Token. Beware, that bits are indexed (bit order) from least significant bit to most significant bit (also called "right to left") while bytes are indexed (byte order) in their natural incrementing byte order (usually written for display purpose from left to write). Endianness does not apply here because each status value fits within a single byte.

Implementations are RECOMMENDED to verify correctness using the test vectors given by this specification.

## Security Guidance for JWT and CWT

Status List Token in JWT format should follow the security considerations of {{RFC7519}} and the best current practices of {{RFC8725}}.

> Status List Token in CWT format should follow the security considerations of {{RFC8392}}.

Reviewer 2: ```suggestion
A Status List Token in the CWT format should follow the security considerations of {{RFC8392}}.
```
